controllers/controllers/resource/fetcher.go

```diff
@@ -617,6 +617,16 @@ func MapMachineTemplateToCloudStackMachineConfigSpec(csMachineTemplate *cloudsta
 		Id:   csMachineTemplate.Spec.Spec.Spec.Template.ID,
 		Name: csMachineTemplate.Spec.Spec.Spec.Template.Name,
 	}
+	csSpec.Spec.DiskOffering = anywherev1.CloudStackResourceDiskOffering{
```

Comment:
don't we need a nil check here for `csMachineTemplate.Spec.Spec.Spec.DiskOffering`?